Tests/MusicXMLTests/Complex Types/ScorePartTests.swift

```diff
@@ -21,4 +21,35 @@ class ScorePartTests: XCTestCase {
         let expected = ScorePart(id: "P1", name: PartName(value: "MusicXML Part"))
         XCTAssertEqual(decoded, expected)
     }
+
+    func testDecoding_complex() throws {
+        // Adopted from MuseScore export
+        let xml = """
+        <score-part id="P1">
+            <part-name>Piano</part-name>
+            <part-abbreviation>Pno.</part-abbreviation>
+            <score-instrument id="P1-I1">
+                <instrument-name>Piano</instrument-name>
+            </score-instrument>
+            <midi-device id="P1-I1" port="1"></midi-device>
+            <midi-instrument id="P1-I1">
+                <midi-channel>1</midi-channel>
+                <midi-program>1</midi-program>
+                <volume>78.7402</volume>
+                <pan>0</pan>
+            </midi-instrument>
+        </score-part>
+        """
+
+        let decoded = try XMLDecoder().decode(ScorePart.self, from: xml.data(using: .utf8)!)
+        let expected = ScorePart(
+            id: "P1",
+            name: PartName(value: "Piano"),
+            partAbbreviation: PartName(value: "Pno."),
+            scoreInstrument: [ScoreInstrument(id: "P1-I1", instrumentName: "Piano")]
+            // TODO: midi-channel and midi-program are not decoding properly
```

Comment:
Is this only held up because a custom `CodingKeys` implementation is needed (e.g., `midiDevice` should be `midi-device`, etc.)?

I have considered configuring the `XMLDecoder` / `XMLEncoder` with the `convertFromKebabCase` and `convertToKebabCase` key decoding / encoding strategies.

This may treat many of the needs for custom `CodingKeys` values.

Reply:
Ah, I understand the problem better now.

Reply:
In the specifications in [score.mod:172](https://github.com/w3c/musicxml/blob/gh-pages/schema/score.mod#L172) the `midi-channel` and `midi-program` are a list of 0 or more tuples `(midi-device?, midi-instrument?)*`.

The `ScorePart.Midi` struct encapsulates this tuple but nothing is provided on how to decode this tuple into this `Midi` struct. Do you have any insights on the XMLCoder on how decoding tuples like this are achieved?